### ISSUE TITLE
Dates to Unix Timestamp

### ISSUE THREAD

**Issue by ryanwjackson** [bbatsov/rubocop#2860]
I can't find a way to satisfy Rubocop and convert dates to unix timestamps.  For time, you can do something like `Time.current.to_i`.  For dates, I have `Date.today.to_time.to_i`.  Rubocop complains about the `to_time` aspect as `Date` does not know anything about TimeZones.

Is there a way around this without disabling date checks?

**Comment by rrosenblum** [comment]
What is your configuration for this cop?

From the documentation for this class
```ruby
      # This cop checks for the correct use of Date methods,
      # such as Date.today, Date.current etc.
      #   
      # Using Date.today is dangerous, because it doesn't know anything about
      # Rails time zone. You must use Time.zone.today instead.
      #   
      # The cop also reports warnings when you are using 'to_time' method,
      # because it doesn't know about Rails time zone either.
      #   
      # Two styles are supported for this cop. When EnforcedStyle is 'strict'
      # then the Date methods (today, current, yesterday, tomorrow)
      # are prohibited and the usage of both 'to_time'
      # and 'to_time_in_current_zone' is reported as warning.
      #   
      # When EnforcedStyle is 'flexible' then only 'Date.today' is prohibited
      # and only 'to_time' is reported as warning.
      #   
      # @example
      #   # no offense
      #   Time.zone.today
      #   Time.zone.today - 1.day
      #   
      #   # acceptable
      #   Date.current
      #   Date.yesterday
      #   
      #   # always reports offense
      #   Date.today
      #   date.to_time
      #   
      #   # reports offense only when style is 'strict'
      #   date.to_time_in_current_zone
```

It looks like the cop is wanting you to use `Time.zone.now.to_i` and `Time.zone.today`. Based on the config, `to_time_in_current_zone` will work if not set to `strict`. I am not sure what the preferred way to convert `Date` to `Time` is.

**Comment by ryanwjackson** [comment]
So the code it doesn't like is in a conditional:

```ruby
val = send(sattr)
if val.is_a?(Date)
  val = val.to_time.to_i # This is the line it hates.
elsif val.is_a?(Time) || val.is_a?(DateTime)
  val = val.to_i
end
```

**Comment by rrosenblum** [comment]
I need someone else to weigh in on this because I am unsure what the accepted way of converting `Date` to `Time` is. The way I interpret the documentation in the cop is, under `strict` style it is never OK to convert `Date` to `Time`. Meaning that all occurrences of `Date` would need to be refactored out. This would be impossible to do because the value could be coming from a gem. If your configuration is not set to `strict`, `val.to_time_in_current_zone` will not register an offense.

@palkan appears to have authored this cop, maybe he can shed some light on the issue.

**Comment by palkan** [comment]
The issue exists. PR attached.

**Comment by rrosenblum** [comment]
Thanks for the PR
